libcudacxx/include/cuda/std/__mdspan/mdspan.h

```diff
@@ -108,28 +110,45 @@ class mdspan
       return (__self.rank() > 0)
           && __MDSPAN_FOLD_OR((__self.__mapping_ref().extents().template __extent<_Idxs>() == index_type(0)));
     }
+
+    template <class... _SizeTypes>
+    _CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI static constexpr bool
+    __check_index(_Extents const& exts, _SizeTypes... __indices)
+    {
+      index_type __zero = 0; // avoid warning:186 (ICC/NVCC) pointless comparison of unsigned integer with zero
+      array<bool, sizeof...(_SizeTypes)> __res{((is_signed_v<index_type> && static_cast<index_type>(__indices) < __zero)
+                                                || static_cast<index_type>(__indices) >= exts.extent(_Idxs))...};
+      _LIBCUDACXX_UNUSED_VAR(__zero * 2);
+      return _CUDA_VSTD::all_of(__res.begin(), __res.end(), [](bool __v) {
+        return __v;
+      });
+    }
```

Comment:
This will instantiate one lambda per extents. We can simply pass cuda::std::identity